sympy/solvers/solveset.py

```diff
@@ -540,6 +540,7 @@ def _solveset(f, symbol, domain, _check=False):
     # _check controls whether the answer is checked or not
 
     from sympy.simplify.simplify import signsimp
+    from sympy.solvers.diophantine import diop_solve
```

Comment:
`Diophantine` is more general that `diop_solve` -- it does pre-conditioning on the equation of interest.